File: smallpict-php/src/Exceptions/PermissionDeniedException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class PermissionDeniedException extends SmallPictException
{
    /**
     * @param string $message
     * @param string|null $requestId
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Permission denied for this endpoint scope',
        ?string $requestId = null,
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, 'FORBIDDEN', 403, $requestId, $details, $previous);
    }
}

File: smallpict-java/src/main/java/com/smallpict/errors/PermissionDeniedException.java
package com.smallpict.errors;

import java.util.Map;

public class PermissionDeniedException extends SmallPictException {
    public PermissionDeniedException(String message) {
        this(message, null, null);
    }

    public PermissionDeniedException(String message, String requestId, Map<String, Object> details) {
        super(message, "FORBIDDEN", 403, requestId, details, null);
    }
}
